pkg/cleaner/azure/delegatedns.go

```diff
@@ -0,0 +1,105 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2017-10-01/dns"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	dnsFailureError    = "Temporary failure in name resolution"
+	e2eterraformPrefix = "e2eterraform"
+	resourceGroup      = "root_dns_zone_rg"
+	zoneName           = "azure.gigantic.io"
+)
+
+func (c Cleaner) cleanDelegateDNSRecords(ctx context.Context) error {
+	var lastError error
+
+	recordsIter, err := c.dnsRecordSetsClient.ListAllByDNSZoneComplete(ctx, resourceGroup, zoneName, nil, "")
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deadLine := time.Now().Add(-gracePeriod).UTC()
```

Comment:
Where is `gracePeriod` coming from?

Reply:
It's defined in another file, resourcegroup.go